enclave/src/rpc/rpc_api_calls.rs

```diff
@@ -0,0 +1,234 @@
+        return Ok((RpcInfo::from(s), false, DirectRequestStatus::Error));
+    }
+
+    let cancel_order_call_args = match verified_trusted_operation.unwrap() {
+        TrustedOperation::direct_call(tcs) => match tcs.call {
+            TrustedCall::cancel_order(a, o, p) => Ok((a, o, p)),
+            _ => Err(RpcCallStatus::operation_type_mismatch),
+        },
+        _ => Err(RpcCallStatus::operation_type_mismatch),
+    };
+
+    if let Err(e) = cancel_order_call_args {
+        return Ok((RpcInfo::from(e), false, DirectRequestStatus::Error));
+    }
+
+    // TODO call implementation here
+
+    Ok((
+        RpcInfo::from(RpcCallStatus::operation_success),
+        false,
+        DirectRequestStatus::Ok,
+    ))
+}
+
+fn withdraw(request: Request) -> RpcResult<(RpcInfo, bool, DirectRequestStatus)> {
+    println!("entering withdraw RPC");
+
+    let verified_trusted_operation = get_verified_trusted_operation(request);
+    if let Err(s) = verified_trusted_operation {
+        return Ok((RpcInfo::from(s), false, DirectRequestStatus::Error));
+    }
+
+    let withdraw_call_args = match verified_trusted_operation.unwrap() {
+        TrustedOperation::direct_call(tcs) => match tcs.call {
+            TrustedCall::withdraw(a, c, b, p) => Ok((a, c, b, p)),
+            _ => Err(RpcCallStatus::operation_type_mismatch),
+        },
+        _ => Err(RpcCallStatus::operation_type_mismatch),
+    };
+
+    if let Err(e) = withdraw_call_args {
+        return Ok((RpcInfo::from(e), false, DirectRequestStatus::Error));
+    }
+
+    // TODO call implementation here
+
+    Ok((
+        RpcInfo::from(RpcCallStatus::operation_success),
+        false,
+        DirectRequestStatus::Ok,
+    ))
+}
+
+fn get_balance(request: Request) -> RpcResult<(RpcInfo, bool, DirectRequestStatus)> {
+    println!("entering get_balance RPC");
+
+    let verified_trusted_operation = get_verified_trusted_operation(request);
+    if let Err(s) = verified_trusted_operation {
+        return Ok((RpcInfo::from(s), false, DirectRequestStatus::Error));
+    }
+
+    let get_balance_call_args = match verified_trusted_operation.unwrap() {
+        TrustedOperation::direct_call(tcs) => match tcs.call {
+            TrustedCall::get_balance(a, c, p) => Ok((a, c, p)),
+            _ => Err(RpcCallStatus::operation_type_mismatch),
+        },
+        _ => Err(RpcCallStatus::operation_type_mismatch),
+    };
+
+    if let Err(e) = get_balance_call_args {
+        return Ok((RpcInfo::from(e), false, DirectRequestStatus::Error));
+    }
+
+    // TODO call implementation here
+
+    Ok((
+        RpcInfo::from(RpcCallStatus::operation_success),
+        false,
+        DirectRequestStatus::Ok,
+    ))
+}
+
+fn get_verified_trusted_operation(request: Request) -> Result<TrustedOperation, RpcCallStatus> {
+    // decode call
+    let shard_id = request.shard;
+    let decode_result = decode_request(request);
+    if let Err(e) = decode_result {
```

Comment:
Just a minor, non-functional imporvement comment: instead of two statements (if let & unwrap) you could use a match statement:
```
let trusted_operation = match decode_request(request) {
 Ok(decoded_result) => top,
 Err(e) => return Err(e),
}
```

Reply:
Thanks, yes that makes it more concise